### PR TITLE
fix(config): reject an mcp_path that panics the server (#653)

### LINKED ISSUE #653: config: malformed mcp_path values panic the server after startup
## Summary

Configuration validation only checks that `mcp_path` starts with `/`. Values that are invalid Go 1.22+ `http.ServeMux` patterns reach `Server.Handler()` and panic the server goroutine.

## Reproduction

```bash
MISTRAL_API_KEY=x DIR2MCP_SKIP_EMBED_PROBE=1 \
  dir2mcp --dir /tmp/corpus --state-dir /tmp/state \
  up --foreground --mcp-path '/{'
```

Observed:

```
panic: parsing "/{": at offset 1: bad wildcard segment (must end with '}')
...
github.com/dirstral/dir2mcp/internal/mcp.(*Server).Handler
```

The process exits with status 2 after other startup work has already occurred.

## Impact

A typo in a persisted config or CLI flag crashes instead of producing the repository's machine-parseable `CONFIG_INVALID` error.

## Acceptance criteria

- Validate/normalize `mcp_path` before binding or starting background work.
- Either restrict it to a safe literal URL-path grammar or safely escape/register it.
- Return `CONFIG_INVALID` with an actionable message for malformed paths.
- Add CLI tests for wildcard/metacharacter, query/fragment, duplicate-slash, and valid custom-path cases.

### PR DESCRIPTION
**Pull request by yidakra**
## The defect

```bash
dir2mcp --dir /tmp/corpus up --foreground --mcp-path '/{'
```

```
panic: parsing "/{": at offset 1: bad wildcard segment (must end with '}')
github.com/dirstral/dir2mcp/internal/mcp.(*Server).Handler
```

Go 1.22's `http.ServeMux` reads `{name}` as a wildcard segment and panics on a malformed one. The only validation was a leading slash, so the value reached `Handler()` and killed the process **after** other startup work had already run. A typo in a persisted config crashed the daemon instead of producing a `CONFIG_INVALID` an operator can read.

After this change, the same command:

```
CONFIG_INVALID: mcp_path "/{" must not contain '{' or '}': Go's request
multiplexer reads those as a wildcard pattern, and a malformed one panics the server
```

Exit code 1, before anything binds.

## Where the gate lives

`Config.Validate()`, not the `up` command. Every entry point runs it: a persisted config, the CLI flag overlay (which #405 added the re-validation for), and `dir2mcp config set`.

## The grammar, and why it is narrow

`mcp_path` names ONE endpoint. Pattern syntax is never useful here, and accepting it would only create ways to write a path that does not mean what it looks like. Each character is refused for its own reason, and the error says which:

| rejected | reason |
|---|---|
| `{` `}` | the panic |
| `?` `#` | not part of the request path, so nothing could ever match |
| `//` | requests are redirected to the cleaned path, so the pattern serves nothing |
| trailing `/` | registers a SUBTREE, so `/mcp/` would also claim `/mcp/anything` |
| whitespace | not a usable path |
| no leading `/` | pre-existing rule, moved here |

Accepted and covered by a test: `/mcp`, `/`, `/deep/nested/mcp`, `/mcp-v2`, `/mcp_v2`, `/MCP`, `/mcp.json`, and empty (the loader supplies the default). A validator that refused a legitimate custom path would break a working deployment, which is worse than the panic it replaces.

## Dead code removed

The leading-slash check in `up` is deleted. `Validate()` runs first at the top of `validateUpConfig`, so that branch became unreachable, and dead code claiming to be a safety net is worse than the duplication.

## Tests

`tests/config/mcp_path_validation_653_test.go`. The last one is the important one:

```go
func TestEveryAcceptedMCPPathRegistersWithoutPanic(t *testing.T) {
	// ... for each value: accepted := cfg.Validate() == nil
	if accepted && registerPanics(path) { t.Errorf("the grammar is too permissive") }
}
```

It asserts the PROPERTY against a real `http.ServeMux` rather than trusting my character list to be complete. A rule that is too permissive fails there even if I missed a case, and it also proves the panic is real rather than assumed.

Proven on main: the headline case plus six of the eight rejected values fail there, and the accepted-value test passes on both.

`make check` passes.

Closes #653

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved MCP path validation to reject malformed or unusable paths before server startup.
  * Prevented invalid paths that could cause request routing failures or startup panics.
  * Continued supporting valid custom, root, nested, and unset MCP paths.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -4634,6 +4635,58 @@ func (c *Config) Validate() error {
...
+	}
+	// `?` and `#` never reach a handler as part of the path: the client sends the
+	// query and the fragment separately, so a path containing them can never
+	// match and the server would silently answer nothing on it.
+	if i := strings.IndexAny(path, "?#"); i >= 0 {
+		return fmt.Errorf("mcp_path %q must not contain %q: a query or fragment is not part of the request path, so no request could ever match it",
+			path, path[i:i+1])
+	}
+	if strings.ContainsAny(path, " \t") {
+		return fmt.Errorf("mcp_path %q must not contain whitespace", path)
```
[the remark this answers is not in the document]
Comment (yidakra) [reply internal/config/config.go]: Correct, and it is worse than reported. Fixed.

You are right that I trimmed a local copy while the server registers `c.MCPPath` untrimmed. Validating something other than the value that is used is the same class of defect this gate exists to close, so this was a live instance of the bug inside the fix for it.

I measured what each whitespace form actually does rather than assume:

```
"/mcp "      -> PANIC: parsing "/mcp ": at offset 0: invalid method "/mcp"
"/mc\tp"    -> PANIC: same reason
"   "        -> PANIC: host/path missing /
" /mcp"      -> registers a route no client can request
"/mc\np"    -> registers a route no client can request
"/mcp\u00a0x" -> registers a route no client can request
```

So a trailing space is a SECOND panic vector I had missed entirely: ServeMux reads a space as the separator before the path, so `/mcp ` parses as the METHOD `/mcp` with no path. My original grammar comment said `{}` was the panic, and that was incomplete.

The non-panicking forms are arguably worse. The daemon starts, the route is registered at an address no client can request, and nothing reports it.

Changes:

- the gate reads `c.MCPPath` directly, no trim;
- every `unicode.IsSpace` rune is refused, not just `' '` and `'\t'`, which is what let the non-breaking space through;
- only an actually EMPTY string still selects the loader default. A whitespace-only value is a typo, and it panics registration rather than meaning "use the default";
- seven regression cases: leading, trailing, internal tab, internal newline, non-breaking space, and two whitespace-only values.

The new cases also join `TestEveryAcceptedMCPPathRegistersWithoutPanic`, which registers every accepted value against a real `http.ServeMux`. That property test is what should have caught this, and it did not only because I left the whitespace family out of its input list. It is in now.
